utils/build/docker/golang/app/internal/rasp/rasp.go

```diff
@@ -0,0 +1,131 @@
+package rasp
+
+import (
+	"encoding/json"
+	"encoding/xml"
+	"errors"
+	"github.com/mattn/go-sqlite3"
+	"gopkg.in/DataDog/dd-trace-go.v1/appsec"
+	"gopkg.in/DataDog/dd-trace-go.v1/appsec/events"
+	"gopkg.in/DataDog/dd-trace-go.v1/contrib/database/sql"
+	httptrace "gopkg.in/DataDog/dd-trace-go.v1/contrib/net/http"
+	"log"
+	"net/http"
+	"os"
+)
+
+func parseRASPRequest(r *http.Request, key string) string {
+	var input string
+
+	switch r.Method {
+	case http.MethodGet:
+		input = r.URL.Query().Get(key)
+	case http.MethodPost:
+		var (
+			err  error
+			body map[string]string
+		)
+		switch r.Header.Get("Content-Type") {
+		case "application/json":
+			err = json.NewDecoder(r.Body).Decode(&body)
+		case "application/xml":
+			var xmlBody []string
+			xml.NewDecoder(r.Body).Decode(&xmlBody)
+			body = map[string]string{key: xmlBody[0]}
+		case "application/x-www-form-urlencoded":
+			err = r.ParseForm()
+			body = map[string]string{key: r.Form.Get(key)}
+		default:
+			err = errors.New("unsupported content type")
+		}
+		if err != nil {
+			log.Fatalf("failed to parse body: %v\n", err)
+		}
+
+		if _, ok := body[key]; !ok {
+			log.Fatalln("missing key in body: ", key)
+		}
+
+		if err := appsec.MonitorParsedHTTPBody(r.Context(), body); err != nil {
+			log.Fatalf("Body Monitoring should not block the request: %v\n", err)
+		}
```

Comment:
Is this really necessary? Can't we leave that to the actual body monitoring tests?